x-pack/plugins/security/public/management/roles/edit_role/privileges/es/index_privilege_form.tsx

```diff
@@ -34,11 +34,12 @@ const toOption = (value: string) => ({ label: value });
 
 interface Props {
   formIndex: number;
-  indexPrivilege: RoleIndexPrivilege;
+  indexType: 'indices' | 'remote_indices';
+  indexPrivilege: RoleIndexPrivilege | RoleRemoteIndexPrivilege;
   indexPatterns: string[];
```

Comment:
Should we ignore the incoming index patterns prop if the indexType is 'remote_indices'? Or will this be superseded by future work where we can populate the actual indices?

Reply:
I think it should be down to the consumer of this component to decide whether they want to pass in this props or not (it's optional). For remote indices we currently don't want to auto-complete local indices so I am not passing the prop in. We might want to auto complete remote indices in the future though so I wouldn't restrict it in the component.